core/src/main/java/org/infinispan/util/concurrent/ConditionFuture.java

```diff
@@ -96,24 +99,31 @@ public void updateAsync(T value, Executor executor) {
 
       try {
          executor.execute(() -> checkConditions(value));
-      } catch (Exception e) {
+      } catch (Throwable t) {
          for (Data data : futures.values()) {
```

Comment:
doesn't need to be synchronized as well? `futures.values()`

Reply:
ps. same for `stop()`

Reply:
Done, good eyes Pedro!